components/flow-orchestration-framework/org.wso2.carbon.identity.flow.engine/src/main/java/org/wso2/carbon/identity/flow/engine/Constants.java

```diff
@@ -0,0 +1,216 @@
+                "Username is not provided in the %s request of flow id: %s"),
+        ERROR_CODE_USERNAME_ALREADY_EXISTS("60003",
+                "Username already exists.",
+                "The provided username already exists in the tenant: %s"),
+        ERROR_CODE_UNDEFINED_FLOW_ID("60004",
+                "Flow id is not defined.",
+                "The flow id is not defined in the %s request."),
+        ERROR_CODE_INVALID_USERNAME("60005",
+                "Invalid username.",
+                "The given username: %s must be an email address."),
+        ERROR_CODE_FLOW_FAILURE("60006",
+                "%s failed.",
+                "%s"),
+        ERROR_CODE_REQUEST_PROCESSING_FAILURE("60007",
+                "Error while processing the %s request.",
+                "%s"),
+        ERROR_CODE_INVALID_USER_INPUT("60008",
+                "Invalid user inputs.",
+                "Invalid user inputs provided for the %s request."),
+        ERROR_CODE_INVALID_ACTION_ID("60009",
+                "Invalid action id.",
+                "The provided action id: %s is invalid."),
+        ERROR_CODE_INVALID_CAPTCHA("60010",
+                "Invalid captcha provided.",
+                "Invalid captcha provided in the %s request of flow id: %s."),
+        ;
+
+        private static final String ERROR_PREFIX = "RFE";
+        private final String code;
+        private final String message;
+        private final String description;
+
+        ErrorMessages(String code, String message, String description) {
+
+            this.code = ERROR_PREFIX + "-" + code;
+            this.message = message;
+            this.description = description;
+        }
+
+        public String getCode() {
+
+            return code;
+        }
+
+        public String getMessage() {
+
+            return message;
+        }
+
+        public String getDescription() {
+
+            return description;
+        }
+
+        @Override
+        public String toString() {
+
+            return code + ":" + message;
+        }
+    }
+
+    /**
+     * Constants for executor status.
+     */
+    public static class ExecutorStatus {
+
+        public static final String STATUS_USER_INPUT_REQUIRED = "USER_INPUT_REQUIRED";
+        public static final String STATUS_EXTERNAL_REDIRECTION = "EXTERNAL_REDIRECTION";
+        public static final String STATUS_INTERACTION = "INTERACTION";
+        public static final String STATUS_USER_CREATED = "USER_CREATED";
+        public static final String STATUS_COMPLETE = "COMPLETE";
+        public static final String STATUS_RETRY = "RETRY";
+        public static final String STATUS_ERROR = "ERROR";
+        public static final String STATUS_USER_ERROR = "USER_ERROR";
+        public static final String STATUS_CLIENT_INPUT_REQUIRED = "CLIENT_INPUT_REQUIRED";
+        // User store manager related statuses.
+        public static final String USER_ALREADY_EXISTING_USERNAME = "UserAlreadyExistingUsername";
```

Comment:
We ideally need to handle this in a different way. Let's have a refactoring after this effort.